video_xblock/backends/wistia.py

```diff
@@ -163,95 +168,101 @@ def get_default_transcripts(self, **kwargs):
-                try:
-                    lang_code = babelfish.Language(lang_code).alpha2
-                except ValueError:
-                    # In case of B or T codes, e.g. 'fre'.
-                    # Reference: https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes
-                    lang_code = babelfish.Language.fromalpha3b(lang_code).alpha2   # pylint: disable=no-member
-                lang_label = self.get_transcript_language_parameters(lang_code)[1]
-                transcript_url = 'url_can_not_be_generated'
-                default_transcript = {
-                    'lang': lang_code,
-                    'label': lang_label,
-                    'url': transcript_url,
-                    'text': text,
-                }
-                default_transcripts.append(default_transcript)
-                self.default_transcripts.append(default_transcript)
-        # If captions do not exist for a video, the response will be an empty JSON array.
-        # Reference: https://wistia.com/doc/data-api#captions_index
-        elif data.status_code == 200 and not wistia_data:
-            message = 'For now, video platform doesn\'t have any timed transcript for this video.'
+        if data.status_code == status.HTTP_200_OK:
+            try:
+                wistia_data = json.loads(data.text)
+            except ValueError:
+                wistia_data = False
+
+            if wistia_data:
+                transcripts_data = [
+                    [el.get('language'), el.get('english_name'), el.get('text')]
+                    for el in wistia_data]
+                # Populate default_transcripts
+                for lang_code, lang_label, text in transcripts_data:
+                    # lang_code, fetched from Wistia API, is a 3 character language code as specified by ISO-639-2.
+                    # Reference: https://wistia.com/doc/data-api#captions_show
+                    # Convert from ISO-639-2 to ISO-639-1; reference: https://pythonhosted.org/babelfish/
+                    try:
+                        lang_code = babelfish.Language(lang_code).alpha2
+                    except ValueError:
+                        # In case of B or T codes, e.g. 'fre'.
+                        # Reference: https://en.wikipedia.org/wiki/List_of_ISO_639-2_codes
+                        lang_code = babelfish.Language.fromalpha3b(lang_code).alpha2   # pylint: disable=no-member
+                    lang_label = self.get_transcript_language_parameters(lang_code)[1]
+                    transcript_url = 'url_can_not_be_generated'
+                    default_transcript = {
+                        'lang': lang_code,
+                        'label': lang_label,
+                        'url': transcript_url,
+                        'text': text,
+                    }
+                    default_transcripts.append(default_transcript)
+                    self.default_transcripts.append(default_transcript)
```

Comment:
It is looks weird, to do same for local variable x and for self.x, where x = default_transcripts. Can you elaborate please? 
Also, it is confusing to have:
default_transcript type dict
default_transcriptS type list
Please consider renaming default_transcript to default_transcript_dict